operator/handler/schedule_test.go

```diff
@@ -184,6 +184,33 @@ func TestScheduleHandler_mergeBackendWithDefaults(t *testing.T) {
 	}
 }
 
+func TestScheduleHandler_mergePodSecurityContextWithDefaults(t *testing.T) {
+	var nilPodSecurityContextPointer *corev1.PodSecurityContext
+	tests := map[string]struct {
+		givenSchedulePodSecurityContext *corev1.PodSecurityContext
+		givenResourcePodSecurityContext *corev1.PodSecurityContext
+		expectedPodSecurityContext      *corev1.PodSecurityContext
+	}{
+		"Given_ResourcePSC_When_SpecPSC_Is_Nil_Then_UseResourcePSC": {
+			givenSchedulePodSecurityContext: nilPodSecurityContextPointer,
+			givenResourcePodSecurityContext: newPodSecurityContext(0, 1, corev1.SELinuxOptions{Level: "s0:c123,c456"}),
+			expectedPodSecurityContext:      newPodSecurityContext(0, 1, corev1.SELinuxOptions{Level: "s0:c123,c456"}),
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			schedule := ScheduleHandler{schedule: &k8upv1.Schedule{Spec: k8upv1.ScheduleSpec{
+				PodSecurityContext: tt.givenSchedulePodSecurityContext,
+			}}}
+			res := &k8upv1.RunnableSpec{
+				PodSecurityContext: tt.givenResourcePodSecurityContext,
+			}
+			schedule.mergeSecurityContextWithDefaults(res)
+			assert.ObjectsAreEqual(*tt.expectedPodSecurityContext, *tt.givenResourcePodSecurityContext)
```

Comment:
that's wrong. There's no `t` passed through and the return value of `ObjectsAreEqual` is a boolean that you don't consume. This results in the test passing even if the objects aren't equal, meaning the test would be a worthless no-op ;)
```suggestion
			assert.Equal(t, *tt.expectedPodSecurityContext, *tt.givenResourcePodSecurityContext)
```

Reply:
Hmm interesting, I expected that it would test the equality. Anyway that's ok, I will change it

Reply:
Well it does test the equality, but it doesn't fail the test because `t` is used to tell the test environment that something failed. And that isn't part of the method signature in `ObjectsAreEqual`